zkevm-circuits/src/mpt_circuit/helpers.rs

```diff
@@ -0,0 +1,1382 @@
+                max_degree,
+                cell_manager,
+                Some(challenges.clone().unwrap().lookup_input().expr()),
+            ),
+            challenges,
+        }
+    }
+
+    pub(crate) fn set_use_dynamic_lookup(&mut self, use_dynamic_lookup: bool) {
+        self.base.set_use_dynamic_lookup(use_dynamic_lookup);
+    }
+
+    pub(crate) fn push_condition(&mut self, condition: Expression<F>) {
+        self.base.push_condition(condition)
+    }
+
+    pub(crate) fn pop_condition(&mut self) {
+        self.base.pop_condition()
+    }
+
+    pub(crate) fn query_bool(&mut self) -> Cell<F> {
+        self.base.query_bool()
+    }
+
+    pub(crate) fn query_byte(&mut self) -> Cell<F> {
+        self.base.query_one(MptCellType::LookupByte)
+    }
+
+    pub(crate) fn query_bytes<const N: usize>(&mut self) -> [Cell<F>; N] {
+        self.base
+            .query_cells_dyn(MptCellType::LookupByte, N)
+            .try_into()
+            .unwrap()
+    }
+
+    pub(crate) fn query_bytes_dyn(&mut self, count: usize) -> Vec<Cell<F>> {
+        self.base.query_cells_dyn(MptCellType::StoragePhase1, count)
+    }
+
+    pub(crate) fn query_cell(&mut self) -> Cell<F> {
+        self.base.query_default()
+    }
+
+    pub(crate) fn query_cells<const N: usize>(&mut self) -> [Cell<F>; N] {
+        self.base
+            .query_cells_dyn(MptCellType::default(), N)
+            .try_into()
+            .unwrap()
+    }
+
+    pub(crate) fn require_equal(
+        &mut self,
+        name: &'static str,
+        lhs: Expression<F>,
+        rhs: Expression<F>,
+    ) {
+        self.base.require_equal(name, lhs, rhs)
+    }
+
+    pub(crate) fn require_in_set(
+        &mut self,
+        name: &'static str,
+        value: Expression<F>,
+        set: Vec<Expression<F>>,
+    ) {
+        self.base.require_in_set(name, value, set)
+    }
+
+    pub(crate) fn require_boolean(&mut self, name: &'static str, value: Expression<F>) {
+        self.base.require_boolean(name, value)
+    }
+
+    pub(crate) fn add_dynamic_lookup(
+        &mut self,
+        description: &'static str,
+        tag: MptCellType,
+        values: Vec<Expression<F>>,
+    ) {
+        self.base.add_dynamic_lookup(description, tag, values)
+    }
+
+    pub(crate) fn add_lookup(
+        &mut self,
+        description: &'static str,
+        cell_type: MptCellType,
+        values: Vec<Expression<F>>,
+    ) {
+        self.base.add_lookup(description, cell_type, values)
+    }
+
+    pub(crate) fn store_dynamic_table(
+        &mut self,
+        description: &'static str,
+        tag: MptCellType,
+        values: Vec<Expression<F>>,
+    ) {
+        self.base.store_dynamic_table(description, tag, values)
+    }
+}
+
+/// Returns `1` when `value == 0`, and returns `0` otherwise.
+#[derive(Clone, Debug, Default)]
+pub struct IsEmptyTreeGadget<F> {
+    is_in_empty_trie: IsEqualGadget<F>,
+    is_in_empty_branch: IsEqualGadget<F>,
```

Comment:
@Brechtpd Not that important, but perhaps we could use some other name instead of `is_in_empty_trie` and `is_in_empty_branch`? Seems a bit strange to be something in the empty something.

Reply:
Any suggestions? For me it kind of makes sense, but I can understand how it could be confusing, though I can't immediately come up with a different name.

Reply:
It seems to me it would be easier to read if we would have:
```
pub struct IsPlaceholderLeafGadget<F> {
    is_empty_trie: IsEqualGadget<F>,
    is_nil_in_branch_at_mod_index: IsEqualGadget<F>,
}
```
instead of
```
pub struct IsEmptyTreeGadget<F> {
    is_in_empty_trie: IsEqualGadget<F>,
    is_in_empty_branch: IsEqualGadget<F>,
}
```

And `is_placeholder_leaf` instead of `is_in_empty_trie` in `account_leaf.rs` and `storage_leaf.rs`.

What do you think?